packages/analytics/amplify_analytics_pinpoint/lib/src/analytics_plugin_impl.dart

```diff
@@ -14,16 +16,25 @@ import 'package:meta/meta.dart';
 class AmplifyAnalyticsPinpoint extends AmplifyAnalyticsPinpointDart {
   /// {@macro amplify_analytics_pinpoint.analytics_plugin_impl}
   AmplifyAnalyticsPinpoint({
-    @visibleForTesting super.endpointInfoStore,
+    @visibleForTesting SecureStorageInterface? endpointInfoStore,
     @visibleForTesting CachedEventsPathProvider? pathProvider,
     @visibleForTesting AppLifecycleProvider? appLifecycleProvider,
     @visibleForTesting DeviceContextInfoProvider? deviceContextInfoProvider,
+    @visibleForTesting LegacyNativeDataProvider? legacyNativeDataProvider,
```

Comment:
I don't see this used in tests. Is there any reason to expose it?

You could add some unit tests that mock this, and then assert that certain behavior does/doesn't happen when the mock does/doesn't provide data, but that should be doable from within amplify_analytics_pinpoint_dart. I don't see a reason to expose it here.

On the subject of tests, It would be a good idea to add tests if possible. We don't have a good way to integ test this, but it seems like some unit tests here would be fairly straightforward and valuable.

Reply:
This is out of scope of this PR, but is there a reason to expose `pathProvider` or `deviceContextInfoProvider`? I don't see those used in tests either.

Reply:
Only exposing necessary fields now.  Other fields that aren't necessary for unit tests have had their @visibleForTesting annotation removed and are no longer present as parameters.